javaslang-test/src/main/java/javaslang/test/Gen.java

```diff
@@ -162,6 +162,27 @@
     }
 
     /**
+     * Chooses an enum value from all the enum constants defined in the enumerated type.
+     * @param clazz Enum class
+     * @return A new enum generator
+     */
+    static <E extends Enum<E>> Gen<E> choose(Class<E> clazz) {
+        return random -> Gen.choose(clazz.getEnumConstants()).apply(random);
+    }
+
+    /**
+     * Chooses a value from all values in the array.
+     * @param values array with the values to choose from
+     * @return A new enum generator
+     */
+    static <E> Gen<E> choose(E[] values) {
```

Comment:
Thx!

Adding a `<T> Gen<T> choose(Iterable<T> values)` would scale best. We could use java.util collections and javaslang collections.

I would implement it similar to `static <T> Gen<T> oneOf(Iterable<Gen<T>> generators)` but using the new `choose(E[])`